code/modules/mob/living/simple_animal/hostile/megafauna/onestarboss.dm

```diff
@@ -0,0 +1,178 @@
+/mob/living/simple_animal/hostile/megafauna/one_star
+	name = "Type - 0315"
+	desc = "Love and concrete."
+
+	faction = "hive"
+
+	icon = 'icons/mob/64x64.dmi'
+	icon_state = "onestar_boss_unpowered"
+	icon_living = "onestar_boss_unpowered"
+	icon_dead = "onestar_boss_wrecked"
+	pixel_x = -16
+	ranged = TRUE
+
+	health = 1700
+	maxHealth = 1700
+	break_stuff_probability = 95
+	stop_automated_movement = 1
+
+	melee_damage_lower = 10
+	melee_damage_upper = 20
+	megafauna_min_cooldown = 30
+	megafauna_max_cooldown = 60
+
+/mob/living/simple_animal/hostile/megafauna/one_star/death()
+	..()
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/LoseTarget()
+	stance = HOSTILE_STANCE_IDLE
+	target_mob = null
+	icon_state = initial(icon_state)
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/LostTarget()
+	stance = HOSTILE_STANCE_IDLE
+	icon_state = initial(icon_state)
+	walk(src, 0)
+
+/mob/living/simple_animal/hostile/megafauna/one_star/FindTarget()
+
+	var/atom/T = null
+	var/valueofview = 0
+
+	if(src.loc:luminosity > 0)
```

Comment:
more : crap
also assuming loc is a turf, which isn't always true (these mobs even have handling for breaking out of non-turf locs)
also very unsure that turf's luminosity is even meaningful, in dynamic lighting systems it's not used, you should use `get_lumcount()`